src/transformers/modeling_utils.py

```diff
@@ -2087,6 +2092,31 @@ def _autoset_attn_implementation(
                     ' We recommend to just use `attn_implementation="flash_attention_2"` when loading the model.'
                 )
 
+            if config._attn_implementation == "use_kernel":
```

Comment:
I would allso support for example passing a repo _id directly no? We can get the device ourselves, this way its easy to just use 1 arg for simplicity!

Reply:
We can create: 
```python
            "cuda": LayerRepository(
                repo_id="medmekk/triton-llama-mlp",
                layer_name="TritonLlamaMLP",
            )
``` 
base on the cls name for example

Reply:
we can have a func wrapper in integration / hub kernels or something

Reply:
And if repo is not supper we juste use eager + warning!